src/Path.hs

```diff
@@ -330,6 +332,41 @@ dirname (Path l) = Path (last (FilePath.splitPath l))
 fileExtension :: Path b File -> String
 fileExtension = FilePath.takeExtension . toFilePath
 
+-- | Add extension to given file path. Throws if the
+-- resulting filename does not parse.
+--
+-- >>> addFileExtension "symbols" $(mkRelFile "Data.List")
+-- "Data.List.symbols"
+-- >>> addFileExtension "evil/" $(mkRelFile "Data.List")
+-- *** Exception: InvalidRelFile "Data.List.evil/"
+--
+-- @since 0.6.1
+addFileExtension :: MonadThrow m
+  => String            -- ^ Extension to add
+  -> Path b File       -- ^ Old file name
+  -> m (Path b File)   -- ^ New file name with the desired extension added at the end
```

Comment:
Why is the argument order different than in [filepath](https://hackage.haskell.org/package/filepath-1.4.1.2/docs/System-FilePath-Posix.html#v:addExtension)? I think I prefer filepath's version.